nomad/resource_job.go

```diff
@@ -409,6 +410,10 @@ func resourceJobRegister(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("error applying jobspec: %s", err)
 	}
 
+	if !d.IsNewResource() {
+		d.Partial(false)
```

Comment:
This is after the job is successfully registered, but before its deployment has succeeded (assuming `detach = false`).  I suppose this makes sense, because the resource is a job, not a deployment, but I guess I want to flag that a failing deployment would still show "no changes" on a subsequent apply, right? So in that case, one would need to `taint` or otherwise destroy and re-apply the job, if the fix for the failing deployment was something other than the job resource config.

Reply:
That's right, but that's Nomad's behaviour. The goal of the TF provider is to update the job state in Nomad. Once that's complete Nomad is responsible for the deployment. Applying again has not effect the same way as running `nomad job run` would also not have any effect.